#**Tugas Praktikum Natural Language Processing**

#**Nama : Gavrilla Claudia**

#**NIM : 21110004**

#**Kelas : S1SD02A**

#**Load Dataset Menjadi Dataframe**

In [1]:
import pandas as pd
from sklearn.feature_extraction.text import TfidfVectorizer

# Memuat dataset dari file CSV
df = pd.read_csv('/content/data.csv')
df

Unnamed: 0                         userName  \
0             0                      Gadri Bandu   
1             1         070_Putri Azzahra Nurdin   
2             2              Novrealita Setiyana   
3             3                       auah gelap   
4             4                      Adi Nugroho   
..          ...                              ...   
995         995                       TEGUH TEAM   
996         996                         Zen Flow   
997         997                     Edward Gosal   
998         998                        Zaaa Qirr   
999         999  Andik Naufal Gaming from Nabila   

                                               content  score  
0    Saya kasih bintang tiga dulu soalnga gemenya b...      3  
1    Bagus bgtt. Dari segi story, grafik sampe puzz...      5  
2    Sejauh ini aku main aku suka banget sama model...      5  
3    Semuanya sudah sempurna tapi Sangat disayangka...      3  
4    Grafik bagus,story bagus,terutama archon quest...      5  
..                                                 ...    ...  
995  Masih bingung sih dengan cara game ini ngerend...      5  
996  game nya seru dari segi story. saran buat Hoyo...      5  
997  Game ini sangat bagus dengan map yg sangat lua...      5  
998  Sangat seru dan asik untuk dimainkan grafik hd...      5  
999  Memuat game terlalu lama . Mau masuk game aja ...      2  

[1000 rows x 4 columns]

**Interpretasi output :**

**Dataset ini terdiri dari 1000 baris dan 4 kolom yaitu "unnamed", "userName", "content", dan "score". Kolom "unnamed" adalah nomor baris dalam DataFrame yang diberikan secara default. Kolom "userName" berisikan nama pengguna dari individu yang memberikan ulasan tentang game. Kolom "content" berisikan ulasan atau komentar yang diberikan oleh pengguna tentang pengalaman mereka dengan game. Kolom "score" skor atau peringkat yang diberikan oleh pengguna untuk game tersebut.**

#**Mengubah kolom "content" menjadi vector menggunakan TFIDFVectorizer dan menampilkan hasilnya**

In [2]:
# Inisialisasi TfidfVectorizer
tfidf_vectorizer = TfidfVectorizer()

# Mengubah kolom "content" menjadi vektor TF-IDF
tfidf_matrix = tfidf_vectorizer.fit_transform(df['content'])

In [3]:
# Menampilkan DataFrame vektor TF-IDF
print(tfidf_matrix)

  (0, 3251)	0.1055299758616986
  (0, 2602)	0.14500522566335713
  (0, 538)	0.18749785600261215
  (0, 340)	0.22117246882497477
  (0, 953)	0.05838791659996207
  (0, 1680)	0.06532245345690199
  (0, 2368)	0.12440423861415989
  (0, 1785)	0.135243996694897
  (0, 3502)	0.12713065282236005
  (0, 1862)	0.09833987824244249
  (0, 1341)	0.1718400527197851
  (0, 639)	0.12951658229423796
  (0, 3145)	0.12312688906052045
  (0, 3440)	0.1718400527197851
  (0, 2802)	0.15059373755015762
  (0, 1915)	0.3998523073106945
  (0, 370)	0.19992615365534724
  (0, 443)	0.22117246882497477
  (0, 870)	0.058161701894011326
  (0, 2147)	0.16625154083298463
  (0, 428)	0.08481675158152252
  (0, 2560)	0.17867983848571975
  (0, 3355)	0.20874417117223965
  (0, 4099)	0.1756559033334547
  (0, 2933)	0.05691042801487717
  :	:
  (997, 1364)	0.06318278374092187
  (997, 377)	0.14013407953934043
  (998, 329)	0.472267532819263
  (998, 1558)	0.472267532819263
  (998, 1058)	0.472267532819263
  (998, 3674)	0.2908434965194849
  (998, 3488)

**Interpretasi output :**

**(0, 3251)	0.1055299758616986 mengindikasikan bahwa kata atau fitur yang terindeks pada posisi 3251 dalam vektor TF-IDF memiliki bobot TF-IDF sekitar 0.1055299758616986 dalam dokumen dengan nomor indeks 0.**

**(997, 1364)	0.06318278374092187 mengindikasikan bahwa kata atau fitur yang terindeks pada posisi 1364 dalam vektor TF-IDF memiliki bobot TF-IDF sekitar 0.06318278374092187 dalam dokumen dengan nomor indeks 999. Demikian seterusnya.**

#**Menampilkan nama fiturnya mengunakan get_features_name_out**

In [4]:
# Mendapatkan daftar nama fitur
feature_names = tfidf_vectorizer.get_feature_names_out()

# Menampilkan daftar nama fitur
print(feature_names)

['00' '000' '000an' ... 'zhong' 'zhongli' 'zonk']


In [5]:
# Membuat DataFrame dari vektor TF-IDF dengan nama fitur
tfidf_df = pd.DataFrame(tfidf_matrix.toarray(), columns=feature_names)
# Menampilkan DataFrame vektor TF-IDF
print("Matriks TFIDF:")
print(tfidf_df)

Matriks TFIDF:
      00  000  000an  08102020  0sebelumnya   10  100  1000  100ms  107  ...  \
0    0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
1    0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
2    0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
3    0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
4    0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
..   ...  ...    ...       ...          ...  ...  ...   ...    ...  ...  ...   
995  0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
996  0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
997  0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
998  0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   
999  0.0  0.0    0.0       0.0          0.0  0.0  0.0   0.0    0.0  0.0  ...   

     yoimiya  you  youtu

**Interpretasi hasil :**

**Dalam matriks ini, terdapat 1000 baris yang mewakili 1000 dokumen atau ulasan berbeda dan 4353 kolom yang mewakili 4353 kata atau token yang diidentifikasi dalam dataset.**

**Dalam matriks ini, sebagian besar nilai adalah 0.0, yang menunjukkan bahwa kata-kata tersebut tidak muncul dalam dokumen yang sesuai dengan dokumen tersebut (nilai TF-IDF = 0).**

In [6]:
# Membuat DataFrame dari vektor TF-IDF dengan nama fitur dan indeks sesuai dengan konten
tfidf_df = pd.DataFrame(tfidf_matrix.toarray(), columns=feature_names, index=df['content'])
# Menampilkan DataFrame vektor TF-IDF
print("Matrik TFIDF:")
print(tfidf_df)

Matrik TFIDF:
                                                     00  000  000an  08102020  \
content                                                                         
Saya kasih bintang tiga dulu soalnga gemenya ba...  0.0  0.0    0.0       0.0   
Bagus bgtt. Dari segi story, grafik sampe puzzl...  0.0  0.0    0.0       0.0   
Sejauh ini aku main aku suka banget sama model ...  0.0  0.0    0.0       0.0   
Semuanya sudah sempurna tapi Sangat disayangkan...  0.0  0.0    0.0       0.0   
Grafik bagus,story bagus,terutama archon quest,...  0.0  0.0    0.0       0.0   
...                                                 ...  ...    ...       ...   
Masih bingung sih dengan cara game ini ngerende...  0.0  0.0    0.0       0.0   
game nya seru dari segi story. saran buat Hoyov...  0.0  0.0    0.0       0.0   
Game ini sangat bagus dengan map yg sangat luas...  0.0  0.0    0.0       0.0   
Sangat seru dan asik untuk dimainkan grafik hd ...  0.0  0.0    0.0       0.0   
Memuat game te

I**nterpretasi hasil :**

**Dari output di atas, terlihat seluruh nilai dalam matriks TF-IDF adalah 0.0 untuk setiap kata atau token dalam setiap dokumen yang ditunjukkan. Ini menunjukkan bahwa dataset yang digunakan mungkin tidak mengandung kata-kata atau token yang ada dalam daftar nama fitur.**

In [7]:
tfidf_df.to_excel("tfidf_matrix.xlsx", index=False)